src/main/java/org/gridsuite/study/server/service/NonEvacuatedEnergyService.java

```diff
@@ -0,0 +1,349 @@
+            .pathSegment(SENSITIVITY_ANALYSIS_API_VERSION, "non-evacuated-energy-results", resultUuidOpt.get().toString())
+            .build().encode().toUri();
+        try {
+            result = restTemplate.getForObject(uri, String.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
+                throw new StudyException(NON_EVACUATED_ENERGY_NOT_FOUND);
+            } else {
+                throw handleHttpError(e, NON_EVACUATED_ENERGY_ERROR);
+            }
+        }
+        return result;
+    }
+
+    public String getNonEvacuatedEnergyStatus(UUID nodeUuid) {
+        String result;
+        Optional<UUID> resultUuidOpt = networkModificationTreeService.getNonEvacuatedEnergyResultUuid(nodeUuid);
+
+        if (resultUuidOpt.isEmpty()) {
+            return null;
+        }
+
+        String path = UriComponentsBuilder.fromPath(DELIMITER + SENSITIVITY_ANALYSIS_API_VERSION + "/non-evacuated-energy-results/{resultUuid}/status")
+            .buildAndExpand(resultUuidOpt.get()).toUriString();
+        try {
+            result = restTemplate.getForObject(sensitivityAnalysisServerBaseUri + path, String.class);
+        } catch (HttpStatusCodeException e) {
+            if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
+                throw new StudyException(NON_EVACUATED_ENERGY_NOT_FOUND);
+            }
+            throw e;
+        }
+        return result;
+    }
+
+    public void stopNonEvacuatedEnergy(UUID studyUuid, UUID nodeUuid) {
+        Objects.requireNonNull(studyUuid);
+        Objects.requireNonNull(nodeUuid);
+
+        Optional<UUID> resultUuidOpt = networkModificationTreeService.getNonEvacuatedEnergyResultUuid(nodeUuid);
+        if (resultUuidOpt.isEmpty()) {
+            return;
+        }
+
+        String receiver;
+        try {
+            receiver = URLEncoder.encode(objectMapper.writeValueAsString(new NodeReceiver(nodeUuid)), StandardCharsets.UTF_8);
+        } catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+        String path = UriComponentsBuilder
+            .fromPath(DELIMITER + SENSITIVITY_ANALYSIS_API_VERSION + "/non-evacuated-energy-results/{resultUuid}/stop")
+            .queryParam(QUERY_PARAM_RECEIVER, receiver).buildAndExpand(resultUuidOpt.get()).toUriString();
+
+        restTemplate.put(sensitivityAnalysisServerBaseUri + path, Void.class);
+    }
+
+    public void invalidateNonEvacuatedEnergyStatus(List<UUID> uuids) {
+        if (!uuids.isEmpty()) {
+            String path = UriComponentsBuilder
+                .fromPath(DELIMITER + SENSITIVITY_ANALYSIS_API_VERSION + "/non-evacuated-energy-results/invalidate-status")
+                .queryParam(RESULT_UUID, uuids).build().toUriString();
+
+            restTemplate.put(sensitivityAnalysisServerBaseUri + path, Void.class);
+        }
+    }
```

Comment:
Why this isn't done at study-server level ?

Reply:
???

Reply:
I mean here we forward the request to sensitivity-analysis-server. I feel like it doesn't make sense to invalidate a result at sensitivity-analysis-server level. For this service the computation does still exist, it is only at study-server level that we want to invalidate a result.